src/Features/VisualBasic/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.vb

```diff
@@ -1,177 +1,124 @@
-            ' could be the last ElseIf statement in a block before an Else statement.
+            Dim relevantIfBlockOrIfStatement As ExecutableStatementSyntax = Nothing
             Dim relevantSpan As TextSpan = Nothing
 
-            If cancellationToken.IsCancellationRequested Then
-                Return
-            End If
-
-            Dim syntaxDocument = Await SyntacticDocument.CreateAsync(document, cancellationToken).ConfigureAwait(False)
-
-            Dim singleLineIf = FindAncestor(Of SingleLineIfStatementSyntax)(syntaxDocument, textSpan.Start, cancellationToken)
-            Dim multiLineIf = FindAncestor(Of MultiLineIfBlockSyntax)(syntaxDocument, textSpan.Start, cancellationToken)
+            Dim singleLineIf = token.GetAncestor(Of SingleLineIfStatementSyntax)()
+            Dim multiLineIf = token.GetAncestor(Of MultiLineIfBlockSyntax)()
 
             If singleLineIf IsNot Nothing AndAlso
                singleLineIf.ElseClause IsNot Nothing Then
 
                 relevantIfBlockOrIfStatement = singleLineIf
-                relevantSpan = singleLineIf.IfKeyword.Span
+                relevantSpan = singleLineIf.Span
 
             ElseIf multiLineIf IsNot Nothing AndAlso
-                   multiLineIf.ElseBlock IsNot Nothing Then
+                   multiLineIf.ElseBlock IsNot Nothing AndAlso
+                   multiLineIf.ElseIfBlocks.IsEmpty Then
 
                 relevantIfBlockOrIfStatement = multiLineIf
-
-                If multiLineIf.ElseIfBlocks.IsEmpty Then
-                    relevantSpan = multiLineIf.IfStatement.IfKeyword.Span
-                Else
-                    Dim elseIfBlock = FindAncestor(Of ElseIfBlockSyntax)(syntaxDocument, textSpan.Start, cancellationToken)
-
-                    ' The MultiLineIfBlockSyntax has ElseIfBlocks and now we want to find out if the
-                    ' user has placed the cursor on the last ElseIfPart or a node contained in the
-                    ' last ElseIfPart to decide whether or not we should provide the code action
-                    If elseIfBlock IsNot Nothing AndAlso
-                       elseIfBlock Is multiLineIf.ElseIfBlocks.Last Then
-
-                        relevantSpan = elseIfBlock.ElseIfStatement.ElseIfKeyword.Span
-                    Else
-                        Return
-                    End If
-                End If
+                relevantSpan = multiLineIf.IfStatement.IfKeyword.Span
             Else
-                Return
+                Return Nothing
             End If
 
             If Not relevantSpan.IntersectsWith(textSpan.Start) Then
-                Return
+                Return Nothing
             End If
 
-            If syntaxDocument.SyntaxTree.OverlapsHiddenPosition(relevantSpan, cancellationToken) Then
-                Return
+            If token.SyntaxTree.OverlapsHiddenPosition(relevantSpan, cancellationToken) Then
+                Return Nothing
             End If
```

Comment:
do we also have a check that we don't offer the feature if the if-node itself spans anything hidden?  if so, we don't really need this check.  (and we should have the total-span check in the base type).

Reply:
not needed here right?  yo uhave the check in the base class.